lib/util/probe.js

```diff
@@ -60,8 +60,56 @@ function observeKafkaStats(msg) {
     kafkaMetrics.observe(JSON.parse(msg.message));
 }
 
+/**
+ * Get probe config will pull the configuration for the probe server based on
+ * the provided site name. If siteNames is empty, it returns the global probe server config
+ * only if it's a single object.
+ *
+ * @param {Object} queueProcessorConfig - Configuration of the queue processor that
+ *      holds the probe server configs for all sites
+ * @param {Array<String>} siteNames - List of site names (should contain at most one element)
+ * @param {Logger} logger - Logger instance
+ * @returns {Object|undefined} Config for site or global config, undefined if no match found or invalid config
+ */
+function getProbeConfig(queueProcessorConfig, siteNames, logger) {
+    if (siteNames.length === 0) {
+        if (!Array.isArray(queueProcessorConfig.probeServer)) {
+            return queueProcessorConfig.probeServer;
+        }
+
+        logger.error('Configuration set for specific sites, but no site provided to the process', {
+            siteNames,
+            queueProcessorConfig,
+        });
+        return undefined;
+    }
+
+    if (Array.isArray(queueProcessorConfig.probeServer)) {
+        if (siteNames.length !== 1) {
+            logger.error('Process configured for more than one site', {
+                siteNames,
+                queueProcessorConfig,
+            });
+            return undefined;
+        }
+
+        const siteConfig = queueProcessorConfig.probeServer.find(config => config.site === siteNames[0]);
+        if (siteConfig === undefined) {
+            logger.warn('Probe server configuration for site not found', {
+                siteName: siteNames[0],
+                queueProcessorConfig,
+            });
+        }
+
+        return siteConfig;
+    }
+
+    return queueProcessorConfig.probeServer;
```

Comment:
I think this could be simplified. We don't actually need the first check. We only care about whether or not the probeserver config is an array.

-> If it's an array -> return the proper config OR undefined when we have multiple sitenames or none.
The way it is deployed in S3C, we always have 1 site name. if there are no sites no process is spawned.

-> Otherwise we just return the probeserver config.
It doesn't matter how many sites we have in this case, as it's the Zenko case, where we can have 0 or N sites and in all cases we only have one config